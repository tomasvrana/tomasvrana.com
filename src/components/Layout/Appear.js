import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Container = styled.div`
display:inline-block;
opacity:0;
&.appear {
  transition:opacity 2s ease;
  opacity:1;
}
`

const Appear = (props) => {
  const [ cls, setCls ] = useState(false)

  useEffect(() => {
    setTimeout(() => {
      setCls(true)
    }, props.delay * 20)
  }, [])

  return (
    <Container className={`${(cls) && 'appear'}`}>
      {props.children}
    </Container>
  )
}

Appear.propTypes = {
  method: PropTypes.string,
  delay: PropTypes.number,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ])
}

export default Appear
