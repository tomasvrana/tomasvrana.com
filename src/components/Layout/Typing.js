import React, { Fragment, useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Typing = (props) => {
  const [ out, setOut ] = useState('')
  const lettersa = 'abcdefghijklmnopqrstuvxyz ,.'
  const letters = 'abcdefghijklmnopqrstuvxyz ,.ěščřžýáíéúů'
  let mystr = props.children
  let typed = ''
  let l = 0
  let p = 0
  let ll = []
  let toggle = true
  let aid
  const arr = letters.split('')
  const myarr = mystr.split('')

  function rollText () {
    if(p < mystr.length){
      toggle = !toggle;
      if (toggle) {
          aid = requestAnimationFrame(rollText);
          return;
      }
      typed = ''
      l++
      if(l >= arr.length){
        l = 0
      }
      if(myarr[p] == arr[l]){
        p++
      }
      for(let i = 0; i < p; i++){
        typed += myarr[i]
      }
      setOut(typed + arr[l])
      aid = requestAnimationFrame(rollText);
    }else{
      if(p == mystr.length){
        setOut(typed)
      }
      cancelAnimationFrame(aid)
    }
  }

  function revealTextSetup () {
    for(let i = 0; i < myarr.length; i++){
      ll[i] = Math.round(Math.random() * myarr.length)
    }
    revealText()
  }

  function revealText () {
    l++
    if(l % 1 == 0){
      const arr = letters.split('')
      const myarr = mystr.split('')
      
      typed = ''
      for(let i = 0; i < myarr.length; i++){
        if(myarr[i].toLowerCase() == arr[ll[i]].toLowerCase()){
          typed += myarr[i]
        }else{
          ll[i]++
          if(ll[i] >= arr.length){
            ll[i] = 0
          }  
          typed += arr[ll[i]]
        }
      }
      setOut(typed)
    }
    aid = requestAnimationFrame(revealText);
  }

  useEffect(() => {
    revealTextSetup()
    return () => {
      //cancelAnimationFrame(aid)
    }
  }, [])

  return (
    <Fragment>
      {out}
    </Fragment>
  )
}

Typing.propTypes = {
  children: PropTypes.children
}

export default Typing
