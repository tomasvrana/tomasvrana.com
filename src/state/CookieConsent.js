import React, { createContext, useState } from 'react'
import PropTypes from 'prop-types'
import { getLocalStorageValue, setLocalStorageValue } from '../helpers/localStorage'
import { allowTracking } from '../helpers/analytics'

const CookieConsentContext = createContext()
const Consumer = CookieConsentContext.Consumer

const Provider = ({ children }) => {
  const [ allowCookies, setAllowCookies ] = useState(getLocalStorageValue('allowCookies') === 'true')

  const updateAllowCookies = allow => {
    if (allow) {
      allowTracking()
      setLocalStorageValue('allowCookies', 'true')
    }

    setAllowCookies(allow)
  }

  return (
    <CookieConsentContext.Provider value={{ allowCookies, setAllowCookies: updateAllowCookies }}>
      {children}
    </CookieConsentContext.Provider>
  )
}

Provider.propTypes = {
  children: PropTypes.any
}

export {
  Provider,
  Consumer
}
