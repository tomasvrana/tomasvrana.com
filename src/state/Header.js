import React, { createContext, useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { getURIPathWithoutLang } from '../helpers/url'

const HeaderContext = createContext()
const Consumer = HeaderContext.Consumer

const pageChangeListeners = []
const onPageChange = () => pageChangeListeners.forEach(listener => listener())

const availableKeys = [
  'byron',
  'shelley',
  'goguen',
  'updates'
]

const getHeaderContentKey = (lang) => {
  const uri = getURIPathWithoutLang(lang)
  const section = uri.replace(/^\//, '').split('/').shift()
  return availableKeys.includes(section) ? section : ''
}

const Provider = ({ children, lang }) => {
  const [ key, setKey ] = useState(getHeaderContentKey(lang))

  const onPageChange = () => setKey(getHeaderContentKey(lang))
  useEffect(() => {
    pageChangeListeners.push(onPageChange)

    return () => pageChangeListeners.pop()
  }, [])

  return (
    <HeaderContext.Provider value={{ key }}>
      {children}
    </HeaderContext.Provider>
  )
}

Provider.propTypes = {
  children: PropTypes.any,
  lang: PropTypes.string.isRequired
}

export {
  Provider,
  Consumer,
  onPageChange
}
