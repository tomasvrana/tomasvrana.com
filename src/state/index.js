import {
  Consumer as CookieConsentConsumer,
  Provider as CookieConsentProvider
} from './CookieConsent'
import {
  Consumer as HeaderConsumer,
  Provider as HeaderProvider
} from './Header'
import {
  Consumer as LanguageConsumer,
  Provider as LanguageProvider
} from './Language'
import {
  Consumer as PageLoaderConsumer,
  Provider as PageLoaderProvider
} from './PageLoader'
import {
  Consumer as ThemeConsumer,
  Provider as ThemeProvider
} from './Theme'

export {
  CookieConsentConsumer,
  CookieConsentProvider,
  HeaderConsumer,
  HeaderProvider,
  LanguageConsumer,
  LanguageProvider,
  PageLoaderConsumer,
  PageLoaderProvider,
  ThemeConsumer,
  ThemeProvider
}
