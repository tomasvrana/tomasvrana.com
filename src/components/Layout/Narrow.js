import React, { Fragment, useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import ReactCSSTransitionGroup from 'react-addons-css-transition-group'
import Head from '../Head'
import PageLoader from '../PageLoader'
import { PageLoaderConsumer } from '../../state'
import { getURIPath } from '../../helpers/url'
import Footer from '../Footer'

const Main = styled.main`
  position: relative;
  z-index: 1;
  margin: 0 1.5rem;
  width: 100%;
  padding:8rem 0 ;
  opacity: 1;
  transition: 0.5s opacity ease-in-out;
  max-width: 650px;

  &.page-content-enter {
    opacity: 0;
  }

  &.page-content-enter-active,
  &.page-content-leave {
    opacity: 1;
  }

  &.page-content-leave-active {
    opacity: 0;
  }
`

const PageLoaderContainer = styled.div`
  opacity: 1;
  transition: 0.5s opacity ease-in-out;

  &.page-content-enter {
    opacity: 0;
  }

  &.page-content-enter-active,
  &.page-content-leave {
    opacity: 1;
  }

  &.page-content-leave-active {
    opacity: 0;
  }
`

const LoadingMask = ({ children, pageLoading }) => {
  const [ showContent, setShowContent ] = useState(null)
  const [ showLoader, setShowLoader ] = useState(null)

  useEffect(() => {
    let timeout
    if (pageLoading) {
      setShowContent(false)
      setShowLoader(true)
    } else {
      setShowLoader(false)
      timeout = setTimeout(() => {
        setShowContent(true)
      }, 500)
    }

    return () => {
      clearTimeout(timeout)
    }
  }, [ pageLoading ])

  return (
    <ReactCSSTransitionGroup
      transitionName='page-content'
      transitionEnterTimeout={500}
      transitionLeaveTimeout={500}
    >
      {((showLoader === null && pageLoading) || showLoader) &&
        <PageLoaderContainer key={`page_loader_${getURIPath()}`}>
          <PageLoader />
        </PageLoaderContainer>
      }
      {((showContent === null && !pageLoading) || showContent) &&
        <Main key={`page_content_${getURIPath()}`}>
          {children}
        </Main>
      }
    </ReactCSSTransitionGroup>
  )
}

LoadingMask.propTypes = {
  children: PropTypes.node,
  pageLoading: PropTypes.bool
}

const Layout = ({ children, headData = null }) => (
  <PageLoaderConsumer>
    {({ loading }) => (
      <Fragment>
        <Head {...headData} />
        <LoadingMask pageLoading={loading}>
          {children}
        </LoadingMask>
        <Footer />
      </Fragment>
    )}
  </PageLoaderConsumer>
)

Layout.propTypes = {
  children: LoadingMask.propTypes.children,
  headData: PropTypes.object
}

export default Layout
