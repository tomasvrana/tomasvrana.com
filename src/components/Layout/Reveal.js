import React, { Fragment, useEffect, useState } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { LanguageConsumer, NavActiveConsumer, ThemeConsumer } from '../../state'

const Container = styled.span`
.blur-1 {
  filter: blur(1px);
}
.blur-2 {
  filter: blur(2px);
}
.blur-3 {
  filter: blur(3px);
}
.blur {
  animation: blink 5s ease 0s infinite;
  -webkit-animation: blink 5s ease 0s infinite;
  -moz-animation: blink 5s ease 0s infinite;
  @keyframes blink {
    0%,20%,40%,60%,80% {
      opacity:0.65
    }
    10%,30%,50%,70%,100% {
      opacity:1
    }
  }
  position: relative;
  &:after {
    content: attr(data-text);
    position: absolute;
    transition:none;
    width: 100%;
    height: 100%;
    opacity:.1;
    top: 0px;
    left: 0;

    animation: inblink .5s ease 0s;
    -webkit-animation: inblink .5s ease 0s;
    -moz-animation: inblink .5s ease 0s;
    @keyframes inblink {
      0%,20%,40%,60%,80% {
        opacity:0
      }
      10%,30%,50%,70%,100% {
        opacity:1
      }
    }
  }
  &.blurOffset-0:after {
    top: 1px;
  }
  &.blurOffset-1:after {
    top: -1px;
  }
  &.blurOffset-2:after {
    top: -5px;
  }
  &.blurOffset-3:after {
    top: -2px;
  }
  &.blurOffset-4:after {
    top: -3px;
  }
  &.blurOffset-5:after {
    top: -4px;
  }
}

span span {
  transition:opacity 8s ease;
}
.glitch{
  animation: glitchblur 3s ease 0s;
  -webkit-animation: glitchblur 3s ease 0s;
  -moz-animation: glitchblur 3s ease 0s;
  @keyframes glitchblur {
    0%,
    80% {
      -webkit-filter: blur(0px);
      -moz-filter: blur(0px);
      -o-filter: blur(0px);
      -ms-filter: blur(0px);
    }
    40% {
      -webkit-filter: blur(1px);
      -moz-filter: blur(1px);
      -o-filter: blur(1px);
      -ms-filter: blur(1px);
    }
  }
}
`

const Box = (props) => {
  const [ blurTimer, setBlurTimer ] = useState([])
  const [ blurOffset, setBlurOffset ] = useState([])
  const [ outArr, setOutArr ] = useState([])
  const [ glitchIndex, setGlitchIndex ] = useState([])
  const [ blurIndex, setBlurIndex ] = useState([])
  const [ blurBool, setBlurBool ] = useState([])
  const [ blurTime, setBlurTime ] = useState([])
  const [ glitchBool, setGlitchBool ] = useState([])
  let letters = `abcdefghijklmnopqrstuvwxyz-@,.' ()ěščřžýáíéúů:0123456789`
  if(props.lang == 'en'){
    letters = `abcdefghijklmnopqrstuvwxyz-@,.' ():0123456789`
  }
  let l = 0
  let bl = []
  let ll = []
  let blurOffsetVar = []
  let block = false
  if(props.children != null){
    block = props.children.split(/\r?\n/)
  }
  let starts = []
  let pause = Math.floor(Math.random() * 1000) + 200
  let glitch = []
  let blur = []
  let blurB = []
  let blurT = []
  let intro = []
  let rolling = []
  let rollingT = []
  let glitchpos = []
  let aid
  const arr = letters.split('')
  const myarr = []

  function revealTextSetup () {
    for(let b = 0; b < block.length; b++){
      myarr[b] = block[b].split('')
      pause = Math.floor(Math.random() * (((myarr[b].length < 100) ? (100 - myarr[b].length) : 1) * 100)) + 100
      glitch[b] = false
      blurB[b] = false
      ll[b] = []
      starts[b] = []
      intro[b] = []
      rolling[b] = []
      rollingT[b] = []
      glitchpos[b] = Math.floor(Math.random() * myarr[b].length)
      blur[b]= Math.floor(Math.random() * myarr[b].length)
      blurT[b] = Math.floor(Math.random() * (((myarr[b].length < 100) ? (101 - myarr[b].length) : 1) * 10))
      bl[b] = 0
      for(let i = 0; i < myarr[b].length; i++){
        ll[b][i] = Math.floor(Math.random() * arr.length)
        starts[b][i] = Math.floor(Math.random() * 100) + 10
        if(props.introVal){
          starts[b][i] = Math.floor(Math.random() * props.introVal) + 10
        }
        intro[b][i] = true
        rolling[b][i] = true
        rollingT[b][i] = Math.floor(Math.random() * 30) + ((myarr[b].length < 100) ? Math.abs(100 - myarr[b].length) : 0)
      }
    }
    setBlurTime(blurT)
    setGlitchIndex(glitchpos)
    setGlitchBool(glitch)
    setBlurBool(blurB)
    revealText()
  }

  function revealText () {
    l++
    let typed = []
    for(let b = 0; b < block.length; b++){
      if(block[b].length > 1 || block[b].length != undefined){
        bl[b]++    
        blurOffsetVar[b] = Math.floor(Math.random() * 5)
        setBlurOffset(blurOffsetVar)

        if(bl[b] == blurT[b]){
          if(blurB[b]){
            blurT[b] = Math.floor(Math.random() * 20) + 3
            bl[b] = 0
            blurB[b] = false
          }else{
            blur[b]= Math.floor(Math.random() * block[b].length)
            blurT[b] = Math.floor(Math.random() * (((myarr[b].length < 100) ? (101 - myarr[b].length) : 1) * 20))
            bl[b] = 0
            blurB[b] = true
          }
        }

        if(l == pause){
          glitch[b] = true
          let newArr = [...glitchBool]
          newArr[b] = true
          setGlitchBool(newArr)
          for(let i = 0; i < block[b].length; i++){
            if(i == glitchpos[b]){
              ll[b][i] = Math.floor(Math.random() * arr.length)
            }
          }
        }
        
        typed[b] = ''
        for(let i = 0; i < block[b].length; i++){
          if(intro[b][i]){
            typed[b] += '_'
            if(l == starts[b][i]){
              intro[b][i] = false;
            }
          }else{
            if(block[b][i].toLowerCase() == arr[ll[b][i]].toLowerCase() && !rolling[b][i]){
              if(block[b][i] != ' '){
                typed[b] += block[b][i]
              }else{
                typed[b] += '='
              }
              if(glitch[b]){
                glitch[b] = false
                pause = Math.floor(Math.random() * (((myarr[b].length < 100) ? (100 - myarr[b].length) : 1) * 100)) + 100
                let newArr = [...glitchBool]
                newArr[b] = false
                setGlitchBool(newArr)
      
                glitchpos[b] = Math.floor(Math.random() * myarr[b].length)
                setGlitchIndex(glitchpos)
                rolling[b][glitchpos[b]] = true
                rollingT[b][glitchpos[b]] = Math.floor(Math.random() * 50) + 5
                l = 0
              }
            }else{
              ll[b][i]++
              if(ll[b][i] >= arr.length){
                ll[b][i] = 0
              }
              if(l > rollingT[b][i]){
                rolling[b][i] = false
              }
              if(glitch[b]){
                if(i == glitchpos[b]){
                  if(block[b][i] == ' '){
                    typed[b] += ' '
                  }else{
                    typed[b] += arr[ll[b][i]]
                  }
                }else{
                  typed[b] += block[b][i]
                }
              }else{
                typed[b] += arr[ll[b][i]]
              }
            }
          }
        }
      }
    }

    let resArr = []
    for(let b = 0; b < typed.length; b++){
      resArr[b] = typed[b].split(' ')
      for(let c = 0; c < resArr[b].length; c++){
        resArr[b][c] = resArr[b][c].split('')
      }
    }
    setOutArr(resArr)
    setBlurTimer(bl)
    setBlurIndex(blur)
    setBlurTime(blurT)
    setBlurBool(blurB)

    aid = requestAnimationFrame(revealText);
  }

  useEffect(() => {
    if(block != false){
      revealTextSetup()
    }
    return () => {
      cancelAnimationFrame(aid)
    }
  }, [])

  return (
    <Container>
      {outArr.map((line, index) => (
        <Fragment>
          {(outArr.length == 1)
          ?
            <span>
              {line.map((word, j) => (
                <span>
                  {word.map((char, h) => (
                    <span data-text={(char == '=') ? ' ' : char} className={`blurOffset-${blurOffset} ${(char == '_') ? 'opacity-0' : '' } ${(glitchIndex[index] == h) ? 'glitch' : ''} ${(blurIndex[index] == h) ? 'blur' : ''}`}>
                      {(char == '=') ? ' ' : char}
                    </span>
                  ))}
                </span>
              ))}
            </span>
          :
            <p>
              {line.map((word, j) => (
                <span>
                  {word.map((char, h) => (
                    <span className={`${(char == '_') ? 'opacity-0' : '' } ${(glitchIndex[index] == h) ? 'glitch' : ''} ${(blurIndex[index] == h) ? 'blur' : ''}`}>
                    {(char == '=') ? ' ' : char}
                    </span>
                  ))}
                </span>
              ))}
            </p>
          }
        </Fragment>
      ))}
    </Container>
  )
}

const Reveal = (props) => {
  return (
    <LanguageConsumer>
      {({ lang }) => (
        <Box lang={lang} introVal={props.introVal}>{props.children}</Box>
      )}
    </LanguageConsumer>
  )
}

Box.propTypes = {
  lang: PropTypes.string,
  introVal: PropTypes.number,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ])
}

Reveal.propTypes = {
  introVal: PropTypes.number,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ])
}

export default Reveal
