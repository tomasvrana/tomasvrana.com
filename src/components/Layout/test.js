import React from 'react'
import Layout from './'
import { shallow } from 'enzyme'
import * as state from '../../state'
import { setPageLoading } from '../../state/PageLoader'

function getWrapper (props) {
  return shallow(
    <Layout {...props}>
      <p>Some content</p>
    </Layout>
  ).dive()
}

describe('<Layout />', () => {
  let context
  beforeEach(() => {
    setPageLoading(true)
    context = { loading: true }
    state.PageLoaderConsumer = jest.fn(props => props.children(context))
  })

  test('it renders correctly when head data is provided', () => {
    expect(getWrapper({
      headData: {
        title: 'Testing'
      }
    })).toMatchSnapshot()
  })

  describe('when page is loading', () => {
    test('it renders correctly', () => {
      expect(getWrapper()).toMatchSnapshot()
    })
  })

  describe('when page is not loading', () => {
    beforeEach(() => {
      setPageLoading(false)
      context = { loading: false }
    })

    test('it renders correctly', () => {
      expect(getWrapper()).toMatchSnapshot()
    })
  })
})
