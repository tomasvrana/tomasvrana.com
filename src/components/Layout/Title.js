import React, { Fragment } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import Reveal from './Reveal'

const PageTitle = styled.h1`
  text-align:left;
  margin:1em 0 2em 0;
  filter: blur(2px);
  text-transform:uppercase;
  &.large {
    text-transform:none;
    font-size:270%;
  }
  animation: blur 16s ease 0s infinite;
  -webkit-animation: blur 16s ease 0s infinite;
  -moz-animation: blur 16s ease 0s infinite;
  @keyframes blur {
    0%,
    39%, 41%, 43%, 45%, 100% {
      -webkit-filter: blur(0px);
      -moz-filter: blur(0px);
      -o-filter: blur(0px);
      -ms-filter: blur(0px);
    }
    40%, 42%, 44%, 46%{
      -webkit-filter: blur(2px);
      -moz-filter: blur(2px);
      -o-filter: blur(2px);
      -ms-filter: blur(2px);
    }
  }
`

const Title = (props) => {
  return (
    <PageTitle className={`${(props.large) ? 'large' : ''}`}>
      <Reveal
        method={props.method}
        rewrite={props.rewrite} 
        rewriteRandom={props.rewriteRandom} 
        rewriteDelay={props.rewriteDelay} 
        delay={props.delay}
      >
        {props.children}
      </Reveal>
    </PageTitle>
  )
}

Title.propTypes = {
  method: PropTypes.string,
  rewrite: PropTypes.string,
  rewriteRandom: PropTypes.string,
  rewriteDelay: PropTypes.string,
  delay: PropTypes.number,
  large: PropTypes.bool,
  children: PropTypes.node
}

export default Title
