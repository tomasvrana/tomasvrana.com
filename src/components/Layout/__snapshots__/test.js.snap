// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> it renders correctly when head data is provided 1`] = `
<Fragment>
  <Layout
    title="Testing"
  />
  <LoadingMask
    pageLoading={true}
  >
    <p>
      Some content
    </p>
  </LoadingMask>
</Fragment>
`;

exports[`<Layout /> when page is loading it renders correctly 1`] = `
<Fragment>
  <Layout />
  <LoadingMask
    pageLoading={true}
  >
    <p>
      Some content
    </p>
  </LoadingMask>
</Fragment>
`;

exports[`<Layout /> when page is not loading it renders correctly 1`] = `
<Fragment>
  <Layout />
  <LoadingMask
    pageLoading={false}
  >
    <p>
      Some content
    </p>
  </LoadingMask>
</Fragment>
`;
